tools/sgartifactregistry/auth.go

```diff
@@ -0,0 +1,68 @@
+package sgartifactregistry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.einride.tech/sage/sg"
+	"go.einride.tech/sage/tools/sggcloud"
+)
+
+// NpmAuthenticate authenticates to the Artifact Registry in the give registryURL.
+// Both npm and yarn are used to configure authentation resulting in an .npmrc file
+// and, in the case of yarn version above 1, a .yarnrc.yml file in the home directory.
+// The npm and yarn commands are run from the give package.json directory in order to
+// use the proper yarn version.
+func NpmAuthenticate(ctx context.Context, packageJSONDir, registryURL string) error {
+	// Grab Google Cloud Access Token
+	var accessTokenOutput strings.Builder
+	cmd := sggcloud.Command(ctx, "auth", "print-access-token")
+	cmd.Stdout = &accessTokenOutput
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	registry := strings.TrimPrefix(registryURL, "https://")
+	// Set npm config
+	cmd = sg.Command(
+		ctx,
+		"npm",
+		"config",
+		"set",
+		"-L",
+		"user",
+		fmt.Sprintf("//%s/:_authToken", registry),
+		strings.TrimSpace(accessTokenOutput.String()),
+	)
```

Comment:
I'm thinking this should do the trick for both NPM and Yarn 1 users

Reply:
Good comment. Changed to a switch statement instead. WDYT?